src/pkg/clients/Amqp.go

```diff
@@ -100,3 +100,131 @@ func (r *amqpClient) connect() error {
 func (r *amqpClient) QueueDeclare() (amqp.Queue, error) {
 	return r.channel.QueueDeclare(r.queueName, false, false, false, false, nil)
 }
+
+type IAmqpExchanger interface {
+	Publish(key string, message []byte) error
+	Consume(q amqp.Queue) (<-chan amqp.Delivery, error)
+	QueueDeclare() (amqp.Queue, error)
+	QueueBind(q amqp.Queue, key string) error
+}
+
+var _ IAmqpExchanger = &amqpExchanger{}
+
+type amqpExchanger struct {
+	channel       *amqp.Channel
+	fullAddr      string
+	exchangerName string
+}
+
+func NewAmqpExchanger(addr, user, pwd, exchangerName string) (IAmqpExchanger, error) {
+	amqpE := &amqpExchanger{
+		channel:       nil,
+		fullAddr:      "amqp://" + user + ":" + pwd + "@" + addr + "/",
+		exchangerName: exchangerName,
+	}
+
+	if err := amqpE.connect(); err != nil {
+		return nil, err
+	}
+
+	if err := amqpE.channel.ExchangeDeclare(
+		amqpE.exchangerName,
+		"direct",
+		true,
+		false,
+		false,
+		false,
+		nil,
+	); err != nil {
+		return nil, err
+	}
+
+	return amqpE, nil
+}
+
+func (r *amqpExchanger) connect() error {
+	amqpConn, err := amqp.Dial(r.fullAddr)
+	if err != nil {
+		return err
+	}
+
+	channel, err := amqpConn.Channel()
+	if err != nil {
+		return err
+	}
+
+	r.channel = channel
+	return nil
+}
+
+func (r *amqpExchanger) Publish(key string, message []byte) error {
+	err := r.channel.Publish(
+		r.exchangerName,
+		key,
+		false,
+		false,
+		amqp.Publishing{
+			ContentType: "text/plain",
+			Body:        message,
+		},
+	)
+
+	if err != nil {
+		if err = r.connect(); err != nil {
+			return err
+		}
+
+		// If we cannot publish, try to reconnect to rabbitMQ service ONE time before
+		// return error
+		// Only consumer should declare queue
+		// if _, err := r.QueueDeclare(); err != nil {
+		// 	return err
+		// }
+
+		return r.channel.Publish(
+			r.exchangerName,
+			key,
+			false,
+			false,
+			amqp.Publishing{
+				ContentType: "text/plain",
```

Comment:
I saw in the futur (or the next PR) that we use json for our publication message and it should be protobuf.

We should only use JSON for our communication front <-> API. For all the other cases, it should be protobuf.